0-module-and-module-source.emu

```diff
@@ -403,13 +434,22 @@ location: https://tc39.es/proposal-compartments/
             </dl>
             <emu-alg>
               1. Assert: _moduleRecord_ is a Module Record.
+              1. Assert: Type(_specifier_) is String.
               1. Assert: _moduleRecord_.[[ModuleInstance]] is an Object.
               1. Assert: ResolveModuleRecordDependency has not been invoked with _moduleRecord_ and _specifier_ pair.
               1. Let _moduleInstance_ be _moduleRecord_.[[ModuleInstance]].
               1. Let _importHook_ be _moduleInstance_.[[ImportHook]].
-              1. Let _completion_ be Completion(Call(_importHook_, *undefined*, « _specifier_ »).
-              1. IfAbruptRejectPromise(_completion_, _promiseCapability_).
-              1. Let _importHookPromise_ be ? PromiseResolve(%Promise%, _completion_.[[Value]]).
+              1. If _importHook_ is *undefined*, then
+                1. NOTE: Default import hook.
+                1. Let _innerPromiseCapability_ be ! NewPromiseCapability(%Promise%).
+                1. TODO: This links/evaluates the imported module, but it should return it in it's initial status (if it was not already linked/evaluated for other reasons).
+                1. Perform HostImportModuleDynamically(_moduleRecord_, _specifier_, _innerPromiseCapability_).
+                1. Let _importHookPromise_ be _innerPromiseCapability_.[[Promise]].
+              1. Else,
+                1. Let _handler_ be _moduleInstance_.[[HandlerValue]].
+                1. Let _completion_ be Completion(Call(_importHook_, _handler_, « _specifier_ »).
+                1. IfAbruptRejectPromise(_completion_, _promiseCapability_).
+                1. Let _importHookPromise_ be ? PromiseResolve(%Promise%, _completion_.[[Value]]).
```

Comment:
```suggestion
                1. Let _importHookPromise_ be Completion(PromiseResolve(%Promise%, _completion_.[[Value]])).
                1. IfAbruptRejectPromise(_importHookPromise_, _promiseCapability_).
```
We must never use `?` in algorithms that signal their success/failure using a promise.